internal/guest/bridge/bridge.go

```diff
@@ -159,19 +163,54 @@ type bridgeResponse struct {
 	response interface{}
 }
 
+type bridgeOpts struct {
+	enableV4 bool
+	cgroupV2 bool
+	host     *hcsv2.Host
+}
+
+type BridgeOpt func(*bridgeOpts) error
+
+func WithV4Enabled(enableV4 bool) BridgeOpt {
+	return func(bo *bridgeOpts) error {
+		bo.enableV4 = enableV4
+		return nil
+	}
+}
+
+func WithCgroupVersion(version uint) BridgeOpt {
+	return func(bo *bridgeOpts) error {
+		if version < 1 || version > 2 {
+			return errors.New(fmt.Sprintf("unsupported CGroup version %d", version))
```

Comment:
Could be `fmt.Errorf("unsupported CGroup version %d", version)` if you need formatting